GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -714,6 +714,24 @@ public bool HasErrors
             {
                 get { return this.ReturnCode != SuccessCode; }

```

Comment:
What do you think about changing this logic to be `==` and the name of this property to be `WasSuccessful` since how it is now is not really and indication that there are errors, it is checking the return code for success?  Then your new logic could be called `HasErrors` and would be more accurate to me.  `HasErrors` sounds like it should be some indication of what is in the `Errors` property which it doesn't now.  I realize this is a lot more invasive of a change but would make more sense at least to me.

Reply:
I considered such a direction, but the scope was larger than I wanted to pick off for this particular bug. It's something we should consider for the backlog, but I'm wary of churning so much of the codebase when the bugfix can be much more targeted. Thoughts @kewillford @sanoursa @wilbaker ?

Reply:
I agree with Kevin that the current naming can make it a bit tricky to follow the current code (see comment in `GitStatusCacheConfig.cs` for an example), but I also agree that we want to limit the scope of this fix.

To throw an option 3 out there, what if for this PR we renamed `HasNonWarningErrors` to something like `HasErrorMessages` and have it simply check `Errors` for lines that don't start with "warning:".


We could then replace calls to `result.Errors.Any()` with `result.HasErrorMessages`:

*Example*

```
if (result.HasErrors)
{
    if (result.HasErrorMessages)
    {
        error = "Error while reading '" + configName + "' from config: " + result.Errors;
```
> should consider for the backlog

I think the cleanup is definitely worth it, whether we do it now or add it to the backlog.

Reply:
This is simplified in the new patch series